### PR TITLE
Rename extentions to extensions

### PR DESCRIPTION
**Pull request by timbussmann**
related to https://github.com/Particular/NServiceBus/pull/3781

### CODE REVIEW

**Comment by danielmarbach** [comment]
Core PR is merged

**Comment by timbussmann** [comment]
the failing tests are mainly related to this ASQ issue: https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/124
if I switch to JSON the following two test fail locally:
* `When_publishing_an_event_implementing_two_unrelated_interfaces`
* `When_started_by_base_event_from_other_saga`

I suggest we move this PR forward and release the beta package as this is required when we push beta4 of the core to nuget. What do you think @Particular/azure-storage-persistence-maintainers ?

**Comment by dbelcham** [comment]
@timbussmann I'm good with this. Can one more @Particular/azure-storage-persistence-maintainers take a look and merge if they're okay with it?

**Comment by Scooletz** [comment]
Sorry @timbussmann  for being picky, but I don't know if we can go with 

> the failing tests are mainly related to this ASQ issue

From the description of the ASQ issue I can tell, that probably the change will be handled by the core. I'd wait for this, then bump up ASQ in ASP and go with the green PR. IMO, if we break tests, it means that we're ok with having red builds.

**Comment by timbussmann** [comment]
maybe it will be handled by the core, but not before beta5 and we still want to get beta4 out. So that means the xml deserialization is fixed in ASQ without changes in the core or it will be fixed with the next beta release.

The tests in ASQ are just green because they don't test the failing scenarios. So just going for green tests obviously isn't the full story.

As I mentioned to all downstream maintainers in the last several days, we want to get beta4 out. If you are concerned that core introduced changes which let these tests fail, I need you to help me investigate here (especially since this repo decided to not use the ATT source packages).

**Comment by timbussmann** [comment]
@Particular/azure-storage-persistence-maintainers ASQ is fixed and tests seem to pass.

File: src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/ConfigureEndpointAzureStorageQueueTransport.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class ConfigureEndpointAzureStorageQueueTransport : IConfigureEndpointTes
     {
         var connectionString = settings.Get<string>("Transport.ConnectionString");
         //connectionString = "UseDevelopmentStorage=true;";
-        configuration.UseSerialization<XmlSerializer>();
+        configuration.UseSerialization<JsonSerializer>();
```
Comment (timbussmann) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/ConfigureEndpointAzureStorageQueueTransport.cs]: this line found a bug (or even more?) in ASQ. So adjusting it (you could also just remove it) just makes us blind to potential issues?

Comment (Scooletz) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/ConfigureEndpointAzureStorageQueueTransport.cs]: I see your point @timbussmann . Reminding that `Xml` is highly unlikely to be used in Azure scenarios.

Comment (timbussmann) [inline src/NServiceBus.Persistence.AzureStorage.AcceptanceTests/ConfigureEndpointAzureStorageQueueTransport.cs]: > Reminding that Xml is highly unlikely to be used in Azure scenarios.

as long as it's not unsupported, I don't think this statement should have any influence on this discussion.

so this is clearly not a fix but a workaround and I don't see any bugs raised/updated? I'm good with moving forward with this PR but then it seems we need to raise an issue about XmlSerializer still broken or reopen. I wasn't involved in the investigation so maybe @dbelcham and @SeanFeldman can chime in?

**Comment by dbelcham** [comment]
@timbussmann based on our troubleshooting yesterday there almost certainly is a bug in ASQ's XML serialization. We didn't go deep enough to figure out what that bug is. We did determine that the XML serialization fails consistently on the Team City build agents, but we couldn't reproduce it locally on developer machines. That led us to believe that there is some difference in the OS, .NET installation (patch/version/??) or some other variable that is affecting the way that the serialization/messaging is happening.

We didn't create a new issue in ASQ for this only because of the question I posed in Slack: do you want to have a new issue or do you want to reopen the last ASQ one that was looking at XML serialization?

**Comment by Scooletz** [comment]
The build error was caused by MyGet problems. It works now.

**Pull request merged**